kb/cluster-metrics-reference.md

```diff
@@ -0,0 +1,148 @@
+
+```
+sum(rate(service_requests{service_type="history", operation="RecordWorkflowTaskStarted"}[2m])) / sum(rate(service_requests{service_type="matching", operation="AddWorkflowTask"}[2m]))
+```
+
+The following list describes some metrics you can get started with.
+
+- `service_requests` - shows service requests received per Task Queue.
+  An example of prometheus query for service requests by operation:
+  `sum(rate(service_requests{operation=\"AddWorkflowTask\"}[2m]))`
+
+- `service_latency` - shows latencies for all Client request operations. Usually these are the starting point to investigate which operation is having a high latency issue.
+  An example of prometheus query for service latency by operation for the Frontend service looks like:
+  `histogram_quantile(0.95, sum(rate(service_latency_bucket{service_type="frontend"}[5m])) by (operation, le))`
+
+- `service_errors_with_type` - (available only in v1.17.0+) to identify errors encountered by the service.
+  An example of prometheus query for service errors by type for the Frontend service could be:
+  `sum(rate(service_errors_with_type{service_type="frontend"}[5m])) by (error_type)`
+
+- `client_errors` - an indicator for connection issues between the different Server roles.
+  An example of Prometheus query for client errors:
+  `sum(rate(client_errors{service_type="frontend",service_role="history"}[5m]))`
+
+Note: All metrics queries in this article are [Prometheus queries](https://prometheus.io/docs/prometheus/latest/querying/basics/).
+The complete list of common metrics is [here](https://github.com/temporalio/temporal/blob/fcec8f99bb199ad28632d5a1c84b9281f816db0c/common/metrics/defs.go#L2496).
+
+In addition to these, you can define some service-specific metrics to get performance details for each service.
+Start with the following list, and use [defs.go](https://github.com/temporalio/temporal/blob/master/common/metrics/defs.go) to define additional metrics as required.
+
+**Matching Service metrics**
+
+- [poll_success_per_tl](https://github.com/temporalio/temporal/blob/6a354ee73f41a26497130f6cd0f4b97ac157f680/common/metrics/defs.go#L2823) - _need description here_
+  Example: `sum(rate(poll_success{}[5m]))`
+- [poll_timeouts](https://github.com/temporalio/temporal/blob/6a354ee73f41a26497130f6cd0f4b97ac157f680/common/metrics/defs.go#L2824) - _need description here_
+  Example: `sum(rate(poll_timeouts{}[5m]))`
+- [poll_success_sync_per_tl](https://github.com/temporalio/temporal/blob/6a354ee73f41a26497130f6cd0f4b97ac157f680/common/metrics/defs.go#L2825)
+  Example: `sum(rate(poll_success_sync{}[5m]))`
+- [asyncmatch_latency](https://github.com/temporalio/temporal/blob/77be7a96c66e6ca1ffc1f6c56d2779e0264d5dfe/common/metrics/defs.go#L2807) - measures async matched tasks from the time they are created to delivered. The larger this latency the longer tasks are sitting in the queue waiting for your workers to pick them up.
```

Comment:
Already link to this source file above - don't need to do it again for each thing. The source lines could change in the future anyways.